rfc/system/1855-type-metadata.md

```diff
@@ -0,0 +1,128 @@
+# [RR] Typing Metadata
+
+**Authors:**
+
+- @maximsmol
+- @kennyworkman
+
+## 1 Executive Summary
+
+A small patch to introduce flytekit support for python-native type metadata.
+
+## 2 Motivation
+
+Users of flyte wish to have access to arbitrary parameter metadata specified in
+common docstring formats (eg.  ReST, Google, Numpydoc-style, etc.) after a
+workflows has been registered.
+
+There already exists property on the `LiteralType` model for metadata.
+
+The `LiteralType` definition in flyteidl already has a `metadata` field in its
+message definition
+(https://github.com/flyteorg/flyteidl/blob/master/protos/flyteidl/core/types.proto#L89),
+suggesting that support for metadata annotations for flyte types was in the
+works.
+
+We wish to add python-native support for this field by allowing flytekit to
+recognize and parse the use of
+[`typing.Annotation`](https://docs.python.org/3/library/typing.html#typing.Annotated).
+
+This could be to change the presentation of a form input ingesting typed values,
+surface descriptions to users or store indexes to control the order of
+parameters with custom frontends.
```

Comment:
To be crisp clear, I would also like to list the things users won't be able to do with this...

```suggestion
This could be to change the presentation of a form input ingesting typed values,
surface descriptions to users or store indexes to control the order of
parameters with custom frontends.

What it doesn't give users:
1. At runtime, when a task is invoked, the metadata won't be sent to the container. If the container has access to the original code, it will be able to access/load metadata from that. This is particularly important to note for TaskTemplate-based flytekit plugins.
2. ?
```

Reply:
i added your proposed anti-feature, but could not think of anymore. please add as many as you want with your greater familiarity of flyte

Reply:
btw, this seems to be the behavior of the proposed implementation